### PR TITLE
feat: add TLS support

### PR DESCRIPTION
**Pull request by setrofim**
- Add -s/--tls flag to indicate that TLS should be used when connecting to the service
- Add -i/--insecure flag to indicate that server cert checks should be skipped
- Add -E/--ca-cert flag to specify additional CA cert(s) to be used in server cert validation (system CA certs are used by default).

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -185,6 +185,8 @@ github.com/subosito/gotenv v1.4.2 h1:X1TuBLAMDFbaTAChgCBLu3DU3UPyELpnF2jjJ2cz/S8
 github.com/subosito/gotenv v1.4.2/go.mod h1:ayKnFf/c6rvx/2iiLrJUk1e6plDbT3edrFNGqEflhK0=
 github.com/veraison/apiclient v0.2.0 h1:QELvZ+eEfzh9v0ORe9B2UTMpiA7aONHpZIfwSfcRR6s=
 github.com/veraison/apiclient v0.2.0/go.mod h1:LCXFZ3D/tJ3HLAOHUg8bnAKGvgTl53e1ntwdwjVbQ5A=
+github.com/veraison/apiclient v0.2.1-0.20240531100343-8a3a730a1e94 h1:0d7vTs3K9Y4bskTtI3pvkFE0HiSHc4vWA3M6Fc0lWRM=
+github.com/veraison/apiclient v0.2.1-0.20240531100343-8a3a730a1e94/go.mod h1:LCXFZ3D/tJ3HLAOHUg8bnAKGvgTl53e1ntwdwjVbQ5A=
```
Comment (thomas-fossati) [inline go.sum]: it looks like there's a missing `go tidy`

Comment (setrofim) [reply go.sum]: Yup, my bad. Fixed.

File: cmd/root.go [hunk 2]
```diff
@@ -50,32 +56,28 @@ func init() {
...
+	)
+	rootCmd.PersistentFlags().StringArrayP(
+		"ca-cert", "E", nil, "path to a CA cert that will be used in addition to system certs; may be specified multiple times",
+	)
+
+	rootCmd.PersistentFlags().VisitAll(func(flag *pflag.Flag) {
+		cfgName := strings.ReplaceAll(flag.Name, "-", "_")
+		err := viper.BindPFlag(cfgName, flag)
+		cobra.CheckErr(err)
+	})
```
Comment (thomas-fossati) [inline cmd/root.go:80]: nice!

**Review by thomas-fossati — APPROVED** [review]
LGTM, thanks!

File: README.md [hunk 3]
```diff
@@ -166,3 +166,12 @@ policy would be applied when the evidence for that scheme is appraised by the
 verification service:
 
     ./pocli deactivate PSA_IOT
+
+### Note on TLS
+
+If the scheme in the API server URL is HTTPS, `pocli` will attempt to establish
```
Comment (yogeshbdeshpande) [inline README.md]: Please add useTLS boolean usage here!

File: README.md [hunk 4]
```diff
@@ -166,3 +166,12 @@ policy would be applied when the evidence for that scheme is appraised by the
 verification service:
 
     ./pocli deactivate PSA_IOT
+
+### Note on TLS
+
+`-s`/`--tls` flag can be used to enable TLS, in which case system CA certs will
+be used to validate the certificate sent by the server. It is possible to
+disable server certificate validation with `-i`/`--insecure` flag (note that if
+this flag is used, `-s` flag is implied and does not need to also be
```
Comment (yogeshbdeshpande) [inline README.md]:
```suggestion
this flag is used, `-s` flag is implied and does not need to be explicitly specified.
```

**Review by yogeshbdeshpande — APPROVED** [review]
LGTM!

**Pull request merged**